controllers/authorino_controller.go

```diff
@@ -161,6 +165,7 @@ func (r *AuthorinoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 // SetupWithManager sets up the controller with the Manager.
 func (r *AuthorinoReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
+		Owns(&k8sapps.Deployment{}).
```

Comment:
If we do not return in line https://github.com/Kuadrant/authorino-operator/blob/6d0039a5745cc88959e8a04bd23d48c491ee2767/controllers/authorino_controller.go#L157 the reconciler will set the status of the CR to ready in line https://github.com/Kuadrant/authorino-operator/blob/6d0039a5745cc88959e8a04bd23d48c491ee2767/controllers/authorino_controller.go#L162 regardless of the status of the deployment, no?

Reply:
Then, instead of `return ctrl.Result{Requeue: true}, nil `, it should be:
```go
return ctrl.Result{}, updateStatusConditions(logger, authorinoInstance, r.Client, statusNotReady())
```

Anyway, leave it as it is. It is working as expected

Reply:
Added a new status condition `DeploymentNotReady` (message: _'Authorino Deployment resource not ready'_), @eguzki